### PR TITLE
updates for the latest fastapi version

### PR DESCRIPTION
**Pull request by evstratbg**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,12 @@
 All notable changes to this project will be documented in this file.
 Please follow [the Keep a Changelog standard](https://keepachangelog.com/en/1.0.0/).
 
+## [0.1.0] - 2024-01-15
```
Comment (zmievsa) [inline CHANGELOG.md:6]: I see that we never changed any versions here. Can you fix this, please?

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,12 @@
 All notable changes to this project will be documented in this file.
```
Comment (zmievsa) [inline CHANGELOG.md:3]: I see that no tests have changed. Did your tests break upon updating fastapi? If they didn't, then it makes sense to add a test that breaks with these changes.

**Review by zmievsa — CHANGES_REQUESTED** [review]
We'll need a second smaller round of review.

I also see that I'll need to fix this in Cadwyn as well. 

Thanks for noticing and creating a PR!

File: pyproject.toml [hunk 3]
```diff
@@ -30,7 +30,7 @@ readme = "README.md"
 
 [tool.poetry.dependencies]
 python = "^3.10"
-fastapi = ">=0.103.1"
```
[the remark this answers is not in the document]
Comment (evstratbg) [reply pyproject.toml:33]: fixed

File: verselect/middleware.py [hunk 4]
```diff
@@ -53,26 +54,28 @@ async def dispatch(
...
-                dependant=self.version_header_validation_dependant,
-            )
-            values, errors, *_ = solved_result
-            if errors:
-                return self.default_response_class(status_code=422, content=_normalize_errors(errors))
-            api_version = cast(date, values[self.api_version_header_name.replace("-", "_")])
-            self.api_version_var.set(api_version)
-        else:
-            api_version = None
+        async with AsyncExitStack() as async_exit_stack:
```
[the remark this answers is not in the document]
Comment (evstratbg) [reply verselect/middleware.py:57]: fixed

File: CHANGELOG.md (hunk 1 above)
Comment (evstratbg) [reply CHANGELOG.md:6]: fixed

File: CHANGELOG.md (hunk 2 above)
Comment (evstratbg) [reply CHANGELOG.md:3]: >Did your tests break upon updating fastapi
yes, if you just update fastapi, current tests will break

**Review by zmievsa — APPROVED** [review]
All done. Thanks for noticing this and fixing it.

:sparkles:

**Pull request merged**